### PR TITLE
RIA-6433 Added handlers to remove age assessment data if different pa…

### PR DESCRIPTION
**Pull request by ThomasKKC**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6433

### Change description ###
Added handlers to remove age assessment data if different path is chosen (non AAA)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AgeAssessmentDataEditAppealHandler.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        Optional<YesOrNo> isAgeAssessmentAppeal = asylumCase.read(AGE_ASSESSMENT, YesOrNo.class);
+
+        if (isAgeAssessmentAppeal.equals(Optional.of(YES))) {
+            String organisationOnDecisionLetter = asylumCase.read(ORGANISATION_ON_DECISION_LETTER, String.class)
+                    .orElseThrow(() -> new RequiredFieldMissingException("Organisation on decision letter missing"));
+
+            //Clear all 'hscTrust' field
+            if (organisationOnDecisionLetter.equals(OrganisationOnDecisionLetter.LOCAL_AUTHORITY.toString())) {
+                log.info("Clearing Prison details for an IRC Detention centre.");
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AgeAssessmentDataEditAppealHandler.java:69]: Typo on log message.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]
Small typo error.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AgeAssessmentDataEditAppealHandler.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    private void clearLitigationFriendData(AsylumCase asylumCase) {
+        log.info("Clearing Litigation Friend data");
+        asylumCase.clear(LITIGATION_FRIEND_GIVEN_NAME);
+        asylumCase.clear(LITIGATION_FRIEND_FAMILY_NAME);
+        asylumCase.clear(LITIGATION_FRIEND_COMPANY);
+        asylumCase.clear(LITIGATION_FRIEND_CONTACT_PREFERENCE);
+        asylumCase.clear(LITIGATION_FRIEND_EMAIL);
+        asylumCase.clear(LITIGATION_FRIEND_PHONE_NUMBER);
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AgeAssessmentDataEditAppealHandler.java:127]: You need to clear the field - litigationFriend

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

**Review by neha-aggarw — DISMISSED** [review]
Do not merge till 6421.

**Review by neha-aggarw — APPROVED** [review]
DO NOT MERGE till RIA-6421.

**Review by hussain-faruk — APPROVED** [review]

**Pull request merged**